testinfra/modules/environment.py

```diff
@@ -0,0 +1,37 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from testinfra.modules.base import InstanceModule
+
+
+class Environment(InstanceModule):
+    """Get Environment variables
+
+    >>> host.environment()
+    {
+        "EDITOR": "vim",
+        "SHELL": "/bin/bash",
+        [...]
+    }
+    """
+
+    def __call__(self):
+        ret_val = dict(
+            i.split('=') for i in self.check_output('env').rstrip().split('\n')
+        )
```

Comment:
IIRC, `check_output()` already do the `rstrip()`.
Also this behave unexpectedly when using environment variable with a `=` in it (example: export FOO=BAR=ZAZ)

I'd suggest instead:
```return dict(i.split('=', 1) for i in self.check_output('env').splitlines()```

But there still issues with environment variables containing a '\n', for this I'd suggest using the `-0` option of `env` (replacing newlines with null).

Reply:
added in #491